### PR TITLE
Resolve stig boot-loader relevant issues

### PR DESCRIPTION
**Pull request by xingzhou**
[#98978094]
[#98978300]
[#98978312]
[#98978246]

Signed-off-by: Zhang Hua <zhuadl@cn.ibm.com>

### CODE REVIEW

File: stemcell_builder/stages/image_install_grub/apply.sh [hunk 1]
```diff
@@ -78,9 +81,20 @@ then
 GRUB_CMDLINE_LINUX="vconsole.keymap=us net.ifnames=0 crashkernel=auto selinux=0 plymouth.enable=0"
 EOF
 
+  pbkdf2_password=`run_in_chroot ${image_mount_point} "echo -e '${random_password}\n${random_password}' | grub2-mkpasswd-pbkdf2 | grep -Eo 'grub.pbkdf2.sha512.*'"`
```
Comment (mariash) [inline stemcell_builder/stages/image_install_grub/apply.sh]: @xingzhou could you please put comments why we configure password for vcap user?

File: stemcell_builder/stages/image_install_grub/apply.sh [hunk 2]
```diff
@@ -156,5 +170,12 @@ else
   exit 2
 fi
 
+md5_password=`run_in_chroot ${image_mount_point} "if which grub-md5-crypt > /dev/null;then echo -e '${random_password}\n${random_password}' | grub-md5-crypt | grep -e '^\\$1\\$.*'; fi"`
```
Comment (mariash) [inline stemcell_builder/stages/image_install_grub/apply.sh]: @xingzhou and also here:

1) why we need to check if grub-md5-crypt is present, can we just call it?
2) lets put password line when we writing template above (line  150,162)
3) always chmod/chown grub.conf

File: stemcell_builder/stages/image_install_grub/apply.sh (hunk 1 above)
Comment (xingzhou) [inline stemcell_builder/stages/image_install_grub/apply.sh]: @mariash, at present, we can not prevent user from editing the startup menu. In this situation, we generated a random password which we do not even know what the password exactly is, under this situation, the start up menu will be protected.

Comment (cppforlife) [inline stemcell_builder/stages/image_install_grub/apply.sh]: @xingzhou Maria meant to put that comment in the code.

File: stemcell_builder/stages/image_install_grub/apply.sh (hunk 2 above)
Comment (xingzhou) [inline stemcell_builder/stages/image_install_grub/apply.sh]: @mariash, 
(1). grub-md5-crypt is there when grub V1 is installed. in CentOS 7 system, we only install grub V2, which will use another method(edit the 00_header config file), so in CentOS 7, grub-md5-cyprt is not there.

(2). I don't catch you on the comment, would you please make some detail description? thanks

(3). for system installed grub V2, the grub config file is /boot/grub2/grub.cfg not /boot/grub/grub.conf, so in this case, we don't chmod/chown grub.conf

File: bosh-stemcell/spec/support/stemcell_centos_rhel_7_shared_examples.rb [hunk 3]
```diff
@@ -13,16 +13,20 @@
       it { should contain 'selinux=0' }
       it { should contain 'plymouth.enable=0' }
       it { should_not contain 'xen_blkfront.sda_is_xvda=1'}
+
       # single-user mode boot should be disabled (stig: V-38586)
       it { should_not contain 'single' }
-    end
 
-    # GRUB 0.97 configuration (used only on Amazon PV hosts) must have same kernel params as GRUB 2
-    describe file('/boot/grub/grub.conf') do
```
Comment (mariash) [inline bosh-stemcell/spec/support/stemcell_centos_rhel_7_shared_examples.rb]: why was this test removed?

**Comment by xingzhou** [comment]
@mariash, for CentOS and RHEL, they are using grub2 as the grub system, which does not need grub.conf file, so we remove the test cases to test the content of grub.conf;(we removed the grub.conf generation code for CentOS and RHEL in the script)

**Comment by cppforlife** [comment]
@xingzhou after looking over these changes I think there was a reason why grub.conf was being generated as comment states:

> GRUB 0.97 configuration (used only on Amazon PV hosts) must have same kernel params as GRUB 2

i think we should bring it back.

**Pull request merged**